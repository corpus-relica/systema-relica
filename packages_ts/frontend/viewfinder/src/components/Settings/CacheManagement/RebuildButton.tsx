import React from 'react';
import { Button } from '../../ui/button.js';

interface RebuildButtonProps {
  isRebuilding: boolean;
  onRebuild: () => void;
  disabled: boolean;
}

const RebuildButton = ({
  isRebuilding,
  onRebuild,
  disabled
}: RebuildButtonProps) => {
  return (
    <Button
      onClick={onRebuild}
      disabled={disabled}
      className="w-full sm:w-auto"
    >
      {isRebuilding ? 'Rebuilding...' : 'Rebuild Cache'}
    </Button>
  );
};

export default RebuildButton;