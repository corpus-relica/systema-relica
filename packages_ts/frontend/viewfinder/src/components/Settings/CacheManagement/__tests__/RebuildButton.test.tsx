import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import RebuildButton from '../RebuildButton';

describe('RebuildButton', () => {
  const mockOnRebuild = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders with default text when not rebuilding', () => {
    render(
      <RebuildButton 
        isRebuilding={false} 
        onRebuild={mockOnRebuild}
        disabled={false}
      />
    );
    
    expect(screen.getByRole('button')).toHaveTextContent('Rebuild Cache');
    expect(screen.getByRole('button')).not.toBeDisabled();
  });

  it('renders with rebuilding text when rebuilding', () => {
    render(
      <RebuildButton 
        isRebuilding={true} 
        onRebuild={mockOnRebuild}
        disabled={true}
      />
    );
    
    expect(screen.getByRole('button')).toHaveTextContent('Rebuilding...');
    expect(screen.getByRole('button')).toBeDisabled();
  });

  it('calls onRebuild when clicked', () => {
    render(
      <RebuildButton 
        isRebuilding={false} 
        onRebuild={mockOnRebuild}
        disabled={false}
      />
    );
    
    const button = screen.getByRole('button');
    fireEvent.click(button);
    
    expect(mockOnRebuild).toHaveBeenCalledTimes(1);
  });

  it('does not call onRebuild when disabled', () => {
    render(
      <RebuildButton 
        isRebuilding={false} 
        onRebuild={mockOnRebuild}
        disabled={true}
      />
    );
    
    const button = screen.getByRole('button');
    fireEvent.click(button);
    
    expect(mockOnRebuild).not.toHaveBeenCalled();
  });

  it('applies correct CSS classes when rebuilding', () => {
    render(
      <RebuildButton 
        isRebuilding={true} 
        onRebuild={mockOnRebuild}
        disabled={true}
      />
    );
    
    const button = screen.getByRole('button');
    expect(button).toHaveClass('opacity-50', 'cursor-not-allowed');
  });

  it('applies correct CSS classes when enabled', () => {
    render(
      <RebuildButton 
        isRebuilding={false} 
        onRebuild={mockOnRebuild}
        disabled={false}
      />
    );
    
    const button = screen.getByRole('button');
    expect(button).toHaveClass('bg-blue-600', 'hover:bg-blue-700');
  });

  it('takes a snapshot when not rebuilding', () => {
    const { container } = render(
      <RebuildButton 
        isRebuilding={false} 
        onRebuild={mockOnRebuild}
        disabled={false}
      />
    );
    
    expect(container).toMatchSnapshot();
  });

  it('takes a snapshot when rebuilding', () => {
    const { container } = render(
      <RebuildButton 
        isRebuilding={true} 
        onRebuild={mockOnRebuild}
        disabled={true}
      />
    );
    
    expect(container).toMatchSnapshot();
  });

  it('takes a snapshot when disabled', () => {
    const { container } = render(
      <RebuildButton 
        isRebuilding={false} 
        onRebuild={mockOnRebuild}
        disabled={true}
      />
    );
    
    expect(container).toMatchSnapshot();
  });
});