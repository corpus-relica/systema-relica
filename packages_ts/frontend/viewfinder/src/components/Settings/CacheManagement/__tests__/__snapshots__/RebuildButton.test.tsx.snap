// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RebuildButton takes a snapshot when disabled 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-[color,box-shadow] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3 w-full sm:w-auto"
    data-slot="button"
    disabled=""
  >
    Rebuild Cache
  </button>
</div>
`;

exports[`RebuildButton takes a snapshot when not rebuilding 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-[color,box-shadow] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3 w-full sm:w-auto"
    data-slot="button"
  >
    Rebuild Cache
  </button>
</div>
`;

exports[`RebuildButton takes a snapshot when rebuilding 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-[color,box-shadow] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3 w-full sm:w-auto"
    data-slot="button"
    disabled=""
  >
    Rebuilding...
  </button>
</div>
`;
